app/javascript/pages/Collaborators/Incomings/Index.tsx

```diff
@@ -261,19 +277,27 @@ const pendingCollaboratorsFirst = (a: IncomingCollaborator, b: IncomingCollabora
   return 0;
 };
 
-export const IncomingCollaborators = () => {
+export default function IncomingsIndex() {
+  const props = usePage<Props>().props;
   const loggedInUser = useLoggedInUser();
-  const navigation = useNavigation();
-
-  const { collaborators: initialCollaborators, collaborators_disabled_reason } =
-    cast<IncomingCollaboratorsData>(useLoaderData());
+  const { collaborators: initialCollaborators, collaborators_disabled_reason } = props;
 
+  // We rely on Inertia props updates, but for immediate UI feedback on sorting/filtering
+  // we might maintain local state or just let the server sort.
+  // The original component used local state for sorting pending first.
   const [incomingCollaborators, setIncomingCollaborators] = React.useState<IncomingCollaborator[]>(
-    initialCollaborators.sort(pendingCollaboratorsFirst),
+    [...initialCollaborators].sort(pendingCollaboratorsFirst),
   );
+
+  // Update state when props change
+  React.useEffect(() => {
+    setIncomingCollaborators([...initialCollaborators].sort(pendingCollaboratorsFirst));
+  }, [initialCollaborators]);
```

Comment:
Can we just do the sorting server side directly and spare ourselves the complexity here?